### PR TITLE
[web][api][SIMS #770] Ministry User views Assessments

### PR DESCRIPTION
**Pull request by ann-aot**
- Requested Assessments data table with datatable pagination and sorting,
- Assessments History data table with datatable pagination and sorting
- code smell is due to TODO
- **Screenshots**
image

### CODE REVIEW

File: sources/packages/web/src/types/contracts/Assessment.ts [hunk 1]
```diff
@@ -19,3 +23,34 @@ export enum AssessmentStatus {
...
+  // When assessmentWorkflowId is null,
+  // then status is Submitted.
+  Submitted = "Submitted",
+
+  // When assessmentWorkflowId is not null
+  // and assessmentData is null, then status is
+  // InProgress.
+  InProgress = "In Progress",
+
+  // when assessmentWorkflowId is not null
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/types/contracts/Assessment.ts:39]: Noce work with the comments, please ensure that the sentences start with an upper case letter.

File: sources/packages/web/src/components/aest/students/assessment/History.vue [hunk 2]
```diff
@@ -0,0 +1,89 @@
+<template>
+  <v-container>
+    <v-card>
+      <v-container>
+        <p class="category-header-large color-blue">
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/aest/students/assessment/History.vue:5]: Can we please use the BodyHeader component here with the count also, like below:
```html
<body-header
      title="Designation agreements"
      :recordsCount="designations?.length"
      class="m-1"
>
```

File: sources/packages/web/src/composables/useAssessment.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  ScholasticStandingStatus,
+  StudentAppealStatus,
+  AssessmentHistoryStatus,
+} from "@/types";
+
+export function useAssessment() {
+  const mapRequestAssessmentChipStatus = (
+    status: StudentAppealStatus | ScholasticStandingStatus,
+  ): StatusChipTypes => {
+    switch (status) {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/composables/useAssessment.ts:12]: This could be simplified as below:
```typescript
 switch (status) {
  case StudentAppealStatus.Pending:
  case ScholasticStandingStatus.Pending:
    return StatusChipTypes.Warning;
  case StudentAppealStatus.Declined:
  case ScholasticStandingStatus.Declined:
    return StatusChipTypes.Error;
  default:
    return StatusChipTypes.Inactive;
}
```

File: sources/packages/api/src/app.module.ts [hunk 4]
```diff
@@ -166,6 +169,9 @@ import { AppSystemAccessModule } from "./app.system-access.module";
     SINValidationService,
     DesignationAgreementLocationService,
     StudentAssessmentService,
+    StudentScholasticStandingsService,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/app.module.ts:172]: Are these ones needed in the main module? I was under the impression that they would be consumed only under the specific modules. Is there any controller/service here that needs them?

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  constructor(
+    private readonly studentAppealService: StudentAppealService,
+    private readonly studentScholasticStandingsService: StudentScholasticStandingsService,
+    private readonly studentAssessmentService: StudentAssessmentService,
+  ) {
+    super();
+  }
+
+  /**
+   * Controller to get all requested assessments for an student
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:31]: Please replace the "an" with an "a" on "for an student" 😉

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      await Promise.all([
+        this.studentAppealService.getPendingAndDeniedAppeals(applicationId),
+        this.studentScholasticStandingsService.getPendingAndDeniedScholasticStanding(
+          applicationId,
+        ),
+      ]);
+
+    const requestedAssessments = [
+      ...studentAppealQuery.map((appeals) => ({
+        ...appeals,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:55]: Is it not better to be explicit about what is returned here. Even the `PendingAndDeniedAppeals` being already small it can grow in the future.

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    ];
+
+    // sorting and returning
+    return requestedAssessments.sort((first, second) =>
+      first.status > second.status ? -1 : 1,
+    );
+  }
+
+  /**
+   * Controller to get history of assessments for an application,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: I would avoid using the word controller here. This would be a method/endpoint/route inside a controller, IMO.

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+   * @returns AssessmentHistorySummaryDTO list.
+   */
+  @Get("application/:applicationId/history")
+  @ApiOkResponse({
+    description: "assessments history found for the application.",
+  })
+  async getAssessmentHistorySummary(
+    @Param("applicationId") applicationId: number,
+  ): Promise<AssessmentHistory[]> {
+    return this.studentAssessmentService.AssessmentHistorySummary(
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:86]: Please use methods with camelCase naming.

File: sources/packages/api/src/app.module.ts (hunk 4 above)
Comment (ann-aot) [reply sources/packages/api/src/app.module.ts:172]: I was also under the same impression,  but during development, API was throwing me an import error in appModules, that is why I added it. But now, after seeing this comment, I removed and tested it's not complaining about the import, I will update the code. Thanks for that

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts (hunk 6 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:55]: agree (Y)

File: sources/packages/api/src/services/student-appeal/student-appeal.service.ts [hunk 9]
```diff
@@ -9,14 +9,21 @@ import {
...
 
 /**
  * Service layer for Student appeals.
  */
 @Injectable()
 export class StudentAppealService extends RecordDataModelService<StudentAppeal> {
-  constructor(connection: Connection) {
+  constructor(
+    connection: Connection,
+    private readonly studentAppealRequests: StudentAppealRequestsService,
```
Comment (dheepak-aot) [inline sources/packages/api/src/services/student-appeal/student-appeal.service.ts:25]: variable name for service instances can end with service IMO.

File: sources/packages/api/src/app.students.module.ts [hunk 10]
```diff
@@ -21,7 +21,9 @@ import {
   StudentRestrictionService,
   DesignationAgreementLocationService,
   StudentAssessmentService,
+  StudentScholasticStandingsService,
```
Comment (dheepak-aot) [inline sources/packages/api/src/app.students.module.ts:24]: Why is StudentScholasticStandingsService imported here? what is controller/service in students module requiring it?

File: sources/packages/api/src/app.system-access.module.ts [hunk 11]
```diff
@@ -26,6 +29,9 @@ import { LoggerModule } from "./logger/logger.module";
     DisbursementScheduleService,
     SequenceControlService,
     StudentRestrictionService,
+    StudentAppealService,
```
Comment (dheepak-aot) [inline sources/packages/api/src/app.system-access.module.ts:32]: Why is StudentAppealService,StudentAppealRequestsService  and StudentScholasticStandingsService imported here?

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+   * @returns RequestAssessmentSummaryDTO list.
+   */
+  @Get("application/:applicationId/requests")
+  @ApiOkResponse({
+    description: "Requested assessment for the application found.",
+  })
+  async getRequestedAssessmentSummary(
+    @Param("applicationId") applicationId: number,
+  ): Promise<RequestAssessmentSummaryDTO[]> {
+    const [studentAppealQuery, studentScholasticStandingsQuery] =
```
Comment (dheepak-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:45]: The response variables can be renamed more relevantly.

File: sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import {
+  AssessmentTriggerType,
+  ScholasticStandingStatus,
+  StudentAppealStatus,
+} from "../../../database/entities";
+
+export enum AssessmentHistoryStatus {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts:7]: I understood that is terms of the UI it does represents the assessment history but I would be more inclined to name this as `StudentAssessmentStatus` because it would be more toward `StudentAssessment` main status. Thoughts?

File: sources/packages/api/src/app.system-access.module.ts (hunk 11 above)
Comment (ann-aot) [reply sources/packages/api/src/app.system-access.module.ts:32]: i had some imports errors in these files, removing

File: sources/packages/api/src/app.students.module.ts (hunk 10 above)
Comment (ann-aot) [reply sources/packages/api/src/app.students.module.ts:24]: i had some imports errors in these files, removing

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+      })),
+      ...studentScholasticStandingsQuery.map((scholasticStanding) => ({
+        submittedDate: scholasticStanding.submittedDate,
+        status: scholasticStanding.status,
+        triggerType: AssessmentTriggerType.ScholasticStandingChange,
+      })),
+    ];
+
+    // submitted date sorting
+    requestedAssessments = requestedAssessments.sort((first, second) =>
```
Comment (dheepak-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: I am not able to follow here. Why are we sorting in dataset instead of query?

Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: I guess because the data-set is an union of 2 tables. Am I right?

File: sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts (hunk 13 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts:7]: mm, at first I was planning to give assessmnetStatus, then in type, he has an assessment_status, which is irrelevant after your today's PR. but `StudentAssessmentStatus` make sense, i will update
image

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts (hunk 14 above)
Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: If a db view approach can make it simple, may be we can put that as techical debt.

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+export class StudentAppealRequestsService extends RecordDataModelService<StudentAppealRequest> {
+  constructor(connection: Connection) {
+    super(connection.getRepository(StudentAppealRequest));
+  }
+  /**
+   * Creates a 'select' query that could be used in an 'exists' or
+   * 'not exists' where clause.
+   * @param appealStatus student request status that to be searched for.
+   * @param isEqual, this flag decides how the passed status should checked,
+   * eg, if isEqual is true, then the logic will check if the passed status
```
Comment (dheepak-aot) [inline sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:19]: what is passed status?

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts (hunk 14 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: please see the service functions, Request datatable need the combination of student appeals table and scholastic standing table

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts (hunk 15 above)
Comment (ann-aot) [reply sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:19]: passed status is the `appealStatus`. i will update the comment

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts (hunk 14 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: > If a db view approach can make it simple, may be we can put that as techical debt.

Yes, I also would say that it is worth taking a look at the DB view approach. We have this opportunity in the upcoming ticket for pending assessments requests where similar results must be paginated.

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+   * is equal to appealStatus.
+   * if isEqual is false, then the logic will check if the passed status
+   * is not equal to appealStatus.
+   * by default isEqual is true.
+   * @returns 'select' query that could be used in an 'exists' or
+   * 'not exists'.
+   */
+  getExistsAppeals(
```
Comment (dheepak-aot) [inline sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:27]: IMO the method can be renamed as getAppealBystatus or getAppealByStatusAndOperator. I believe our intention is to pass the status and operator(Equals, NotEqual)

File: sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts [hunk 17]
```diff
@@ -0,0 +1,9 @@
+import { ScholasticStandingStatus } from "../../database/entities";
+
+/**
+ * Service model to fetch Pending And Denied scholastic standings.
+ */
+export interface PendingAndDeniedScholasticStandings {
```
Comment (dheepak-aot) [inline sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts:6]: This model represents ScholasticStandings, we can detach the status from the interface name. If it is very basic model then we can prefix it by Basic or something like that IMO.

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts (hunk 16 above)
Comment (ann-aot) [reply sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:27]: as in comment it's used in exist and non exists, we have a lot of examples for subquery, please check and let me know 
image

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+      ...studentScholasticStandings.map((scholasticStanding) => ({
+        submittedDate: scholasticStanding.submittedDate,
+        status: scholasticStanding.status,
+        triggerType: AssessmentTriggerType.ScholasticStandingChange,
+      })),
+    ];
+
+    // submitted date sorting
+    requestedAssessments = requestedAssessments.sort((first, second) =>
+      new Date(first.submittedDate) > new Date(second.submittedDate) ? 1 : -1,
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: `submittedDate` is already a Date, is there any reason to convert it again to a Date?

File: sources/packages/api/src/services/student-assessment/student-assessment.service.ts [hunk 19]
```diff
@@ -349,4 +351,55 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
...
+   * * InProgress.
+   * * WHEN 3:if assessmentWorkflowId is not null
+   * * and assessmentData is not null, then status
+   * * is Completed.
+   * @param applicationId applicationId.
+   * @returns StudentAssessment list
+   */
+  async assessmentHistorySummary(
+    applicationId: number,
+  ): Promise<AssessmentHistory[]> {
```
[the remark this answers is not in the document]
Comment (ann-aot) [reply sources/packages/api/src/services/student-assessment/student-assessment.service.ts:372]: updating the comment, we don't have status in the `assessment` table as mentioned in the comment, with WHEN clause we can get the status from the query directly, no need for extra logic in the controller, also we need to sort the status, 
image

File: sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts [hunk 20]
```diff
@@ -0,0 +1,17 @@
+import {
+  AssessmentTriggerType,
+  ScholasticStandingStatus,
+  StudentAppealStatus,
+} from "../../../database/entities";
+
+export enum StudentAssessmentStatus {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts:7]: This should be declared at the service level.

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+  StudentAppealRequest,
+  StudentAppealStatus,
+} from "../../database/entities";
+
+@Injectable()
+export class StudentAppealRequestsService extends RecordDataModelService<StudentAppealRequest> {
+  constructor(connection: Connection) {
+    super(connection.getRepository(StudentAppealRequest));
+  }
+  /**
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:14]: Please add a blank line above this one.

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+   * Creates a 'select' query that could be used in an 'exists' or
+   * 'not exists' where clause.
+   * @param appealStatus student request status that to be searched for.
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:17]: Please review the text. I believe that the idea was to have something like _student request status to be searched for._.

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+   * @param isEqual, this flag decides how the passed status should checked,
+   * eg, if isEqual is true, then the logic will check if the passed appealStatus
+   * is equal to appealStatus of the selected row..
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:20]: Please remove the extra period.

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts (hunk 16 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:27]: If the method will be changed I would prefer the first suggested option `getAppealByStatus`.

File: sources/packages/api/src/services/student-appeal/student-appeal.model.ts [hunk 24]
```diff
@@ -1,7 +1,16 @@
+import { StudentAppealStatus } from "../../database/entities";
+
 /**
  * Service model for student appeal.
  */
 export interface StudentAppealRequestModel {
   formName: string;
   formData: any;
 }
+/**
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/student-appeal/student-appeal.model.ts:10]: Please add a blank line above this one.

File: sources/packages/api/src/services/student-appeal/student-appeal.service.ts [hunk 25]
```diff
@@ -72,4 +79,53 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
...
+  /**
+   * Get all pending and declined student Appeals
+   * for an application.
+   * Here we have added different when statement
+   * in CASE to fetch the status of the appeals.
+   * * WHEN: is checking if at least one Pending
+   * * appeals request is there for an appeal,
+   * * then the status is considered as Pending.
+   * * END: if any of them is not falling
+   * * under above case and applying  the andWhere
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/student-appeal/student-appeal.service.ts:92]: Thanks for adding the comments explaining the details.
Please remove the extra white spaces in lines 92 and 94.

File: sources/packages/api/src/services/student-assessment/student-assessment.models.ts [hunk 26]
```diff
@@ -0,0 +1,12 @@
+import { StudentAssessmentStatus } from "src/route-controllers/assessment/models/assessment.dto";
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/student-assessment/student-assessment.models.ts]: Please use relative routes.

File: sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts [hunk 27]
```diff
@@ -0,0 +1,37 @@
...
+    return this.repo
+      .createQueryBuilder("scholasticStanding")
+      .select("scholasticStanding.scholasticStandingStatus", "status")
+      .addSelect("scholasticStanding.submittedDate", "submittedDate")
+      .innerJoin("scholasticStanding.application", "application")
+      .where("application.id = :applicationId", { applicationId })
+      .andWhere("scholasticStanding.scholasticStandingStatus != :status", {
+        status: ScholasticStandingStatus.Approved,
+      })
+      .getRawMany();
```
Comment (dheepak-aot) [inline sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts:35]: why getRawMany?

File: sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts (hunk 17 above)
Comment (dheepak-aot) [reply sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts:6]: That is the logic of service to extract only retrieve ScholasticStandings of a certain status. But ScholasticStanding is a model which respresents the transformed entity for service.

File: sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts (hunk 27 above)
Comment (dheepak-aot) [reply sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts:35]: Sorry I will correct my question. Why we are using service model here?

File: sources/packages/web/src/components/generic/StatusChip.models.ts [hunk 28]
```diff
@@ -4,3 +4,8 @@ export enum StatusChipTypes {
   Error = "error",
   Inactive = "inactive",
 }
+
+export enum RequestedAssessmentStatusChipTypes {
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/generic/StatusChip.models.ts:8]: Is this enum used anywhere?

File: sources/packages/web/src/components/aest/students/assessment/History.vue [hunk 29]
```diff
@@ -0,0 +1,89 @@
+<template>
+  <v-container>
+    <v-card>
+      <v-container>
+        <body-header title="History" class="m-1"> </body-header>
+        <span>A history of assessments</span>
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/aest/students/assessment/History.vue:6]: The body header also has a sub-title. Is it what we are looking for here?
image

File: sources/packages/web/src/components/aest/students/assessment/Request.vue [hunk 30]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <v-container>
+    <v-card>
+      <v-container>
+        <body-header title="Requests" class="m-1"> </body-header>
+        <span
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/aest/students/assessment/Request.vue:6]: The body header also has a sub-title. Is it what we are looking for here?

File: sources/packages/web/src/components/aest/students/assessment/History.vue [hunk 31]
```diff
@@ -0,0 +1,89 @@
+<template>
+  <v-container>
+    <v-card>
+      <v-container>
+        <body-header title="History" class="m-1"> </body-header>
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/aest/students/assessment/History.vue:5]: Can we please add the _recordsCount_ also here?

File: sources/packages/web/src/components/aest/students/assessment/Request.vue [hunk 32]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <v-container>
+    <v-card>
+      <v-container>
+        <body-header title="Requests" class="m-1"> </body-header>
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/aest/students/assessment/Request.vue:5]: Can we please add the _recordsCount_ also here?

File: sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts (hunk 16 above)
Comment (ann-aot) [reply sources/packages/api/src/services/student-appeal-request/student-appeal-request.service.ts:28]: updated to `appealsByStatusQueryObject`
image

File: sources/packages/web/src/components/generic/StatusChip.models.ts (hunk 28 above)
Comment (ann-aot) [reply sources/packages/web/src/components/generic/StatusChip.models.ts:8]: no removing

File: sources/packages/web/src/components/aest/students/assessment/History.vue (hunk 31 above)
Comment (ann-aot) [reply sources/packages/web/src/components/aest/students/assessment/History.vue:5]: recount is not in figma, 
image
 i will double check with Lynn

File: sources/packages/web/src/components/aest/students/assessment/History.vue [hunk 33]
```diff
@@ -0,0 +1,89 @@
...
+<script lang="ts">
+import {
+  DEFAULT_PAGE_LIMIT,
+  PAGINATION_LIST,
+  AssessmentHistorySummaryDTO,
+} from "@/types";
+import { ref, onMounted } from "vue";
+import { StudentAssessmentsService } from "@/services/StudentAssessmentsService";
+import { useFormatters } from "@/composables";
+import ContentGroup from "@/components/generic/ContentGroup.vue";
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/components/aest/students/assessment/History.vue:57]: Not part of this PR but we can start importing components like _ContentGroup_ and _BodyHeader_ globally to avoid importing them multiple times 😉

**Review by andrewsignori-aot — COMMENTED** [review]
Really nice work 😉 
Please take a look at the comments.

File: sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts (hunk 13 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/assessment/models/assessment.dto.ts]: Thanks for that. Yes, the `assessment_status` was not named in the most accurate way and it caused issues while creating the table specific for the reassessments. I can see the challenge. Thanks for adjusting it.

File: sources/packages/web/src/components/aest/students/assessment/History.vue (hunk 31 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/aest/students/assessment/History.vue:5]: Please double check but there is no reason to not add them here 😉

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts (hunk 12 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts]: I did not follow why the variables could be considered "not relevantly named".
I saw that they were already renamed, but both ways would work, IMO.

File: sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts (hunk 17 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts:6]: As we talked @ann-aot, I can see why it was first done in this way but the reasons do not exist anymore because of the mapping on the controller hence it would be good t change as @dheepak-aot mentioned.

File: sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts (hunk 27 above)
Comment (andrewsignori-aot) [reply sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts:35]: As we talked @ann-aot, I can see why it was first done in this way but the reasons do not exist anymore because of the mapping on the controller hence it would be good t change as @dheepak-aot mentioned.

File: sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts (hunk 17 above)
Comment (ann-aot) [reply sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.models.ts]: function is updated, please have a look

**Comment by dheepak-aot** [comment]
Great work @ann-aot  👍
Please have a look at the comments.

**Review by guru-aot — APPROVED** [review]
LGTM, Nice work @ann-aot

File: sources/packages/web/src/components/aest/students/assessment/History.vue (hunk 31 above)
Comment (ann-aot) [reply sources/packages/web/src/components/aest/students/assessment/History.vue]: checked with @abschwenker and added the record count

File: sources/packages/web/src/components/aest/students/assessment/Request.vue (hunk 32 above)
Comment (ann-aot) [reply sources/packages/web/src/components/aest/students/assessment/Request.vue]: added

**Review by henryoforeh-aot — APPROVED** [review]

**Review by dheepak-aot — APPROVED** [review]
LGTM @ann-aot 👍Thanks for doing the changes.

File: sources/packages/web/src/main.ts [hunk 34]
```diff
@@ -117,6 +127,8 @@ AppConfigService.shared.init().then(() => {
     .component("TabPanel", TabPanel)
     .component("font-awesome-icon", FontAwesomeIcon)
     .component("Timeline", Timeline)
+    .component("BodyHeader", BodyHeader)
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/main.ts:130]: Nice, thanks!

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes, looks great 👍
